sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala

```diff
@@ -370,8 +371,11 @@ case class Literal (value: Any, dataType: DataType) extends LeafExpression {
         case _ => v + "D"
       }
     case (v: Decimal, t: DecimalType) => v + "BD"
-    case (v: Int, DateType) => s"DATE '${DateTimeUtils.toJavaDate(v)}'"
-    case (v: Long, TimestampType) => s"TIMESTAMP('${DateTimeUtils.toJavaTimestamp(v)}')"
+    case (v: Int, DateType) => s"DATE '${DateFormatter().format(v)}'"
```

Comment:
is this consistent with cast?

Reply:
Yes, `CAST` uses the same: https://github.com/apache/spark/blob/8efc5ec72e2f5899547941010e22c023d6cb86b3/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L242

Reply:
Can we just call `Cast` here?

Reply:
We can do that but here we need `String`. Using `CAST` we will do unnecessary conversion to `UTF8String`: `dateFormatter.format()` produces `String` -> `UTF8String` -> and we will have to convert it back to `String`.

Reply:
makes sense